synapse/types.py

```diff
@@ -404,7 +404,7 @@ def f2(m):
     return username.decode("ascii")
 
 
-@attr.s(frozen=True, slots=True, cmp=False)
+@attr.s(frozen=True, slots=True, order=False)
```

Comment:
I had to fix this to make the `if result.next_batch == since_token` conditional work. As I wrote in #8518, I think it might have been exacerbating the situation.